person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission = 'Unknown')
+    @id = Random.rand(1...1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_services?
+    age >= 18 || @parent_permission == true
+  end
+
+  private
+
+  def of_age?
+    age > 18
```

Comment:
- [ OPTIONAL ] I think here you forget to add age >= 18 since this method Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise. so kindly update it and please here name age as @age because you are working with @age variable ✅💯🔥

Reply:
Thank you so much for your observation @Reem-lab !!! I will make sure to make the change, and thanks a lot for your kind review!!! Have an awesome new year
![ThankYouMinionGIF](https://user-images.githubusercontent.com/50700379/210407830-c2e71bd6-4f4b-49ae-863c-4643b9db609b.gif)
!